doc/source/algorithms.rst

```diff
@@ -163,6 +163,7 @@ The following is an example that shows how to use this function:
 The result has several fields, among which the following are most useful:
 
 * ``parents[i]``:  the parent vertex of the i-th vertex. The parent of each source vertex is itself.
+* ``hasparent[i]``:  ``true`` if the i-th vertex has a parent, and ``false`` otherwise. Note that ``hasparent[i] == true`` for all source vertices.
```

Comment:
I would add some interpretation for the case where ``hasparent[i] == false``. I.e., this means that the vertex isn't reachable from any source.